runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7697,6 +7723,7 @@ J9::X86::TreeEvaluator::VMnewEvaluator(
             dataOffset,
             tempReg,
             monitorSlotIsInitialized,
+            dynamicArrayAllocation,
```

Comment:
If you pass `true` here, then the path in `genInitObjectHeader()` that loads the array type for "old" AOT can be deleted, since we'll start using the path that does it for `dynamicArrayAllocation` instead. Personally I think that would be more straightforward because effectively what old AOT is doing here is forcing the allocation to be dynamic

Reply:
Good point.